roboflow/models/object_detection.py

```diff
@@ -132,27 +133,42 @@ def predict(
             stroke=stroke,
             labels=labels,
         )
-        # Check if image exists at specified path or URL
-        self.__exception_check(image_path_check=image_path)
+
+        # Check if image exists at specified path or URL or is an array
+        if hasattr(image_path, "__len__") == True:
+            pass
+        else:
+            self.__exception_check(image_path_check=image_path)
 
         # If image is local image
         if not hosted:
-            # Open Image in RGB Format
-            image = Image.open(image_path).convert("RGB")
-
-            # Create buffer
-            buffered = io.BytesIO()
-            image.save(buffered, quality=90, format="JPEG")
-            # Base64 encode image
-            img_str = base64.b64encode(buffered.getvalue())
-            img_str = img_str.decode("ascii")
-
-            # Post to API and return response
-            resp = requests.post(
-                self.api_url,
-                data=img_str,
-                headers={"Content-Type": "application/x-www-form-urlencoded"},
-            )
+            if ".jpg" in image_path or ".png" in image_path:  # Open Image in RGB Format
```

Comment:
should we support other formats as well?

Reply:
discussed here as well -> https://github.com/roboflow-ai/roboflow-python/pull/68 (in the comments)

Reply:
Feel free to support other image formats, those were just the ones that have been tested.

Using imencode saves time when you are using OpenCV2 to read data. How we currently have the package coded will not accept OpenCV2 frames because there is no logic to handle Numpy arrays.

By using the image encoder this way we ensure that no one need to write an image on either side of the deployment when using OpenCV2.

![Crashing](https://user-images.githubusercontent.com/113200203/197644668-55f1a35d-8e6d-407a-9135-241eb8c2ceaa.png)

In the scenario above, I will need to write an image to work with the predict function. I would rather pass the cv2 frame array.